### PR TITLE
dm: fix SSRAM region unmap failure issue

### PR DESCRIPTION
**Pull request by yonghuah**
In ACRN HSM driver, only MMIO type is supported
 when doing EPT unmap.

 This patch walk around above issue to unmap
 SSRAM region before doing EPT mapping.

Tracked-On: #7563
Signed-off-by: Yonghua Huang <yonghua.huang@intel.com>

### CODE REVIEW

**Comment by acrnsi** [comment]
Can one of the admins verify this patch?

**Comment by yonghuah** [comment]
@ywan170  This PR is for 3.0 specific, pls review, thanks.

File: devicemodel/hw/platform/vssram/vssram.c [hunk 1]
```diff
@@ -458,10 +457,19 @@ static int vssram_ept_map_buffer(struct vmctx *ctx, struct vssram_buf *buf_desc)
 	memmap.vma_base = buf_desc->vma_base;
 	memmap.user_vm_pa = buf_desc->gpa_base;
 	memmap.len = buf_desc->size;
+
+	/*
+	 * This is a Walk-around:
+	 * - As only MMIO type can be supported when doing EPT unmap in HSM driver,
+	 *   so hardcoding of 'type' to be MMIO type.
+	 */
+	memmap.type = ACRN_MEMMAP_MMIO;
```
Comment (ywan170) [inline devicemodel/hw/platform/vssram/vssram.c:466]: What's the memory type of MMIO EPT page entry? If it is UC, then SSRAM can not work...

Comment (yonghuah) [reply devicemodel/hw/platform/vssram/vssram.c:466]: @ywan170  This MMIO type is for Unmap case only and is a Walk Around solution to by-bass a check in ACRN kernel.
 fix for upstream is still in-progress.

**Review by ywan170 — APPROVED** [review]

**Pull request merged**